src/transaction/log_page_buffer.c

```diff
@@ -2549,6 +2552,32 @@ logpb_write_page_to_disk (THREAD_ENTRY * thread_p, LOG_PAGE * log_pgptr, LOG_PAG
   return NO_ERROR;
 }
 
+/*
+ * logpb_fill_or_reset_header_parameters - initialize values from header, or reset to invalid values if no header
+ *                    argument is provided
+ */
+void
+logpb_fill_or_reset_header_parameters (const struct log_header *hdr, PGLENGTH & io_page_size, PGLENGTH & log_page_size,
```

Comment:
You could go with two different functions: logpb_fill_header_parameters and logpb_reset_header_parameters.

Reply:
Done